integration-tests/src/tests/network/runner.rs

```diff
@@ -485,24 +365,36 @@ pub struct Runner {
     num_nodes: usize,
     num_validators: usize,
     test_config: Vec<TestConfig>,
-    state_machine: Option<StateMachine>,
-
-    info: Option<Arc<RwLock<RunningInfo>>>,
+    state_machine: StateMachine,
 
     accounts_id: Option<Vec<AccountId>>,
     ports: Option<Vec<u16>>,
     validators: Option<Vec<AccountId>>,
     genesis_time: Option<DateTime<Utc>>,
 }
 
+struct NodeHandle {
+    addr: Addr<PeerManagerActor>,
+    send_stop: tokio::sync::oneshot::Sender<()>,
```

Comment:
SG, done. Is that how you imagined it implemented? The code looks more convoluted now IMO. I can't really move stop(), to NodeHandle drop(), because it is asynchronous.

Reply:
Yeah, it's true that going via drop is a bit more convoluted. It's a tradeoff between having to use fancier code, and statically-checking shutdown state machine.